src/main/cc/wfa/virtual_people/common/field_filter/equal_filter.cc

```diff
@@ -0,0 +1,181 @@
+  ValueType value_;
+};
+
+template <>
+class EqualFilterImpl<std::string> : public EqualFilter {
+ public:
+  explicit EqualFilterImpl(
+      std::vector<const google::protobuf::FieldDescriptor*>&& field_descriptors,
+      absl::string_view value):
+      field_descriptors_(std::move(field_descriptors)),
+      value_(value) {}
+
+  bool IsMatch(const google::protobuf::Message& message) const override;
+
+ private:
+  std::vector<const google::protobuf::FieldDescriptor*> field_descriptors_;
+  std::string value_;
+};
+
+template <typename ValueType>
+bool EqualFilterImpl<ValueType>::IsMatch(
+    const google::protobuf::Message& message) const {
+  return value_ == GetValueFromProto<ValueType>(message, field_descriptors_);
+}
+
+template <>
+bool EqualFilterImpl<const google::protobuf::EnumValueDescriptor*>::IsMatch(
+    const google::protobuf::Message& message) const {
+  return (
+      value_->number() ==
+      GetValueFromProto<const google::protobuf::EnumValueDescriptor*>(
+          message, field_descriptors_)->number());
+}
+
+bool EqualFilterImpl<std::string>::IsMatch(
+    const google::protobuf::Message& message) const {
+  return (
+      value_ ==
+      GetValueFromProto<const std::string&>(message, field_descriptors_));
+}
+
+// NumericType can be
+//   int32_t
+//   int64_t
+//   uint32_t
+//   uint64_t
+//   bool
+template <typename NumericType>
+absl::StatusOr<std::unique_ptr<EqualFilterImpl<NumericType>>> CreateFilter(
+    const FieldFilterProto& config,
+    std::vector<const google::protobuf::FieldDescriptor*>&& field_descriptors) {
+  absl::StatusOr<NumericType> value =
+      ConvertToNumeric<NumericType>(config.value());
+  if (!value.ok()) {
+    return absl::InvalidArgumentError(absl::StrCat(
+        "Value type does not match name. Input FieldFilterProto: ",
+        config.DebugString()));
+  }
+  return absl::make_unique<EqualFilterImpl<NumericType>>(
+      std::move(field_descriptors), *value);
+}
+
+absl::StatusOr<std::unique_ptr<EqualFilter>> EqualFilter::New(
+    const google::protobuf::Descriptor* descriptor,
+    const FieldFilterProto& config) {
+  if (config.op() != FieldFilterProto::EQUAL) {
+    return absl::InvalidArgumentError(absl::StrCat(
+        "Op must be EQUAL. Input FieldFilterProto: ", config.DebugString()));
+  }
+  if (!config.has_name()) {
+    return absl::InvalidArgumentError(absl::StrCat(
+        "Name must be set. Input FieldFilterProto: ", config.DebugString()));
+  }
+  if (!config.has_value()) {
+    return absl::InvalidArgumentError(absl::StrCat(
+        "Value must be set. Input FieldFilterProto: ", config.DebugString()));
+  }
+  std::vector<const google::protobuf::FieldDescriptor*> field_descriptors;
+  RETURN_IF_ERROR(
+      GetFieldFromProto(descriptor, config.name(), &field_descriptors));
+
+  switch (field_descriptors.back()->cpp_type()) {
+    case google::protobuf::FieldDescriptor::CppType::CPPTYPE_INT32:
+      {
```

Comment:
Could you delete these unnecessary brackets? No need for this to take up so much vertical space

Reply:
Done.